backends/tfhe-cuda-backend/cuda/src/integer/multiplication.cuh

```diff
@@ -362,9 +359,12 @@ __host__ void host_integer_mult_radix_kb(
     ch_amount = r / chunk_size;
     dim3 add_grid(ch_amount, num_blocks, 1);
     size_t sm_size = big_lwe_size * sizeof(Torus);
-    cuda_memset_async(new_blocks, 0,
-                      ch_amount * num_blocks * big_lwe_size * sizeof(Torus),
-                      stream);
+    if (ch_amount != 0) {
+      cuda_memset_async(
+          new_blocks, 0,
+          ch_amount * num_blocks * big_lwe_size * sizeof(Torus),
```

Comment:
Yeah I know, as num_blocks == 0 should never occur we don't have an early return... same for n = 0. We would have to add those checks in every function basically, right now we just assume we're doing things right and never call those functions with num_blocks == 0 or lwe_dimension == 0.

Reply:
I've added one now, but I'm not really satisfied :sweat_smile: I think it would be better to either implement the early return, or assume everywhere num_blocks >0